db/schema.rb

```diff
@@ -132,6 +132,7 @@
     t.inet "last_sign_in_ip"
     t.boolean "admin", default: false, null: false
     t.integer "group_id"
+    t.datetime "deleted_at"
     t.index ["email"], name: "index_users_on_email", unique: true
```

Comment:
This index may cause trouble. If we delete a user then try to add a user with the same email it will lead to an inconsistency.

We should be able to drop the index and then rebuild it with `WHERE deleted_at IS NOT NULL`.

Reply:
I think the UUID index may need the same treatment. If a user get's deleted, changes their email in login, then gets re-added, that'll break things.

Reply:
Good catch - thank you.

Reply:
Wait...that should be `WHERE deleted_at IS NULL`, right?  We want the index to work on records that "exist", correct?